core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb_test_export.go

```diff
@@ -59,10 +60,33 @@ func (env *testVDBEnv) init(t testing.TB, cache *statedb.Cache) {
 }
 
 // startExternalResource sstarts external couchDB resources for testVDBEnv.
-func (env *testVDBEnv) startExternalResource() {
-	if env.couchAddress == "" {
-		env.couchAddress, env.cleanupCouchDB = couchdbtest.CouchDBSetup(nil)
+func (env *testVDBEnv) startExternalResource(t *testing.T) {
+	if env.couchAddress != "" {
+		return
+	}
+	// check if couchDB is being started externally.
+	externalCouchAddress, isSet := os.LookupEnv("COUCHDB_ADDR")
+	if isSet {
+		env.couchAddress, env.cleanupCouchDB = externalCouchAddress, func() {}
+		return
+	}
+
+	couchDB := &runner.CouchDB{}
+	require.NoError(t, couchDB.Start())
+	env.couchAddress, env.cleanupCouchDB = CouchDBSetup(t, nil)
+}
+
+// CouchDBSetup sets up the CouchDB if it is not running already
+func CouchDBSetup(t *testing.T, binds []string) (addr string, cleanup func()) {
+	// check if couchDB is being started externally.
+	externalCouchAddress, isSet := os.LookupEnv("COUCHDB_ADDR")
+	if isSet {
+		return externalCouchAddress, func() {}
 	}
+
```

Comment:
Same as you removed this at other place, we can get rid of this here as well.

Reply:
This is as per the existing couchdb startup code.